refactor(transformer/async-generator-functions): move handling of `MethodDefinition`'s value to `exit_function`

Dunqing opened the pull request:
Part of #7074

I have mentioned the reason why I need to move this in https://github.com/oxc-project/oxc/pull/7105

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7106?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7106** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7106?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#7105** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7105?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented on crates/oxc_transformer/src/es2018/async_generator_functions/mod.rs:
```diff
@@ -164,22 +164,14 @@ impl<'a, 'ctx> Traverse<'a> for AsyncGeneratorFunctions<'a, 'ctx> {
 
-    fn exit_function(&mut self, _func: &mut Function<'a>, _ctx: &mut TraverseCtx<'a>) {
+    fn exit_function(&mut self, func: &mut Function<'a>, ctx: &mut TraverseCtx<'a>) {
+        if func.r#async && func.generator && !func.is_typescript_syntax() {
+            self.executor.transform_function_for_method_definition(func, ctx);
+        }
```
Do you need to check if this function is a method?

```suggestion
        if func.r#async
            && func.generator
            && !func.is_typescript_syntax()
            && matches!(ctx.parent(), Ancestor::MethodDefinitionValue(_))
        {
            self.executor.transform_function_for_method_definition(func, ctx);
        }
```

Dunqing commented:
### Merge activity

* **Nov 4, 10:43 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
refactor(transformer/async-generator-functions): move handling of `MethodDefinition`'s value to `exit_function` (#7106)

Part of #7074

I have mentioned the reason why I need to move this in https://github.com/oxc-project/oxc/pull/7105
